### PR TITLE
Use bun:test

### PR DESCRIPTION
**Pull request by fel1x-developer**
### The author should do the following, if applicable

`bun:test` is the first-party test runner on bun. It is faster than vitest.

- [ ] Add tests
- [x] Run tests
- [x] `bun run format:fix && bun run lint:fix` to format the code
- [ ] Add [TSDoc](https://tsdoc.org/)/[JSDoc](https://jsdoc.app/about-getting-started) to document the code

### CODE REVIEW

File: runtime-tests/bun/vitest.config.ts
Comment (yusukebe) [inline runtime-tests/bun/vitest.config.ts:1]: This is a breaking change. If we remove this, we can't get the coverage of this Bun runtime test.

Comment (exoego) [reply runtime-tests/bun/vitest.config.ts:1]: As you can see in CodeCov report, branch coverage is decreased a bit due to this change.
If you trust Bun's test, we can switch test runner from Vitest to Bun.

Comment (yusukebe) [reply runtime-tests/bun/vitest.config.ts:1]: @exoego 

Thanks! If the CodeCov result will be changed, I think we don't have to merge this PR.

**Comment by yusukebe** [comment]
Hi @fel1x-developer

As this comment: https://github.com/honojs/hono/pull/3556#discussion_r1816251202, the CodeCov result will be changed with this PR. There is no big reason to switch `vitest` to `bun:test`. We can not merge this now. Thanks.

**Pull request closed without merging**